lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -36,10 +36,16 @@ class GoogleCloudOutput < BufferedOutput
     config_param :private_key_path, :string, :default => nil
     config_param :private_key_passphrase, :string, :default => 'notasecret'
 
-    # If use_metadata_service is set to true, we obtain the project_id, zone,
-    # and vm_id from the GCE metadata service.  Otherwise, those parameters
-    # must be specified in the config file explicitly.
+    # Specify project/instance metadata.
+    #
+    # project_id, zone, and vm_id are required to have valid values, which
+    # can be obtained from the metadata service or set explicitly.
+    # Otherwise, the plugin will fail to initialize.
+    #
+    # Whether to attempt to obtain metadata from the local metadata service.
+    # It is safe to specify 'true' even on platforms with no metadata service.
     config_param :use_metadata_service, :bool, :default => true
```

Comment:
I can expand the comment, but the aim of this CL is basically to make this flag unnecessary by auto-detecting whether we have a metadata service or not (in the next CL we will also auto-detect which platform we are running on).  If there is no metadata service then we will try to fetch from it and fail as you noted below.

We explicitly do not want it to be "all or nothing" because on EC2 we cannot obtain project_id from the metadata service.